fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/LakeTableTieringManager.java

```diff
@@ -0,0 +1,426 @@
+                lock,
+                () ->
+                        // schedule it again if one round of tiering is finished
+                        doHandleStateChange(tableId, TieringState.Scheduled));
+    }
+
+    public void renewTieringHeartbeat(long tableId) {
+        inLock(
+                lock,
+                () -> {
+                    liveTieringTableIds.put(tableId, clock.milliseconds());
+                });
+    }
+
+    /**
+     * Handle the state change of the lake table to be tiered. The core state transitions for the
+     * state machine are as follows:
+     *
+     * <p>New -> Scheduled:
+     *
+     * <p>-- When the lake table is newly created, do: schedule a timer to wait for a tiering
+     * interval configured in table, transmit the table to Pending
+     *
+     * <p>ReInitialized -> Scheduled：
+     *
+     * <p>-- When the coordinator server is restarted, for all existing lake table, do: schedule a
+     * timer to wait for a tiering service timeout, transmit the tables to Pending. Note, we must
+     * wait for the tiering service timeout in case any existing tiering service is tiering the
+     * table
+     *
+     * <p>Tiering -> Scheduled
+     *
+     * <p>-- When the tiering service finished one-round of tiering, do: schedule a timer to wait
+     * for a tiering interval configured in table, transmit the table to Pending
+     *
+     * <p>Scheduled -> Pending
+     *
+     * <p>-- The time interval to wait has passed, do: transmit to Pending state
+     *
+     * <p>Tiering -> Pending
+     *
+     * <p>-- When the tiering service is timeout, do: transmit to Pending state
+     *
+     * <p>Pending -> Tiering
+     *
+     * <p>-- When the table is assigned to a tiering service after tiering service request the
+     * table, do: transmit to Tiering state
+     */
+    private void doHandleStateChange(long tableId, TieringState targetState) {
+        TieringState currentState = tieringStates.get(tableId);
+        if (!isValidStateTransition(currentState, targetState)) {
+            LOG.error(
+                    "Fail to change state for table {} from {} to {} as it's not a valid state change.",
+                    tableId,
+                    currentState,
+                    targetState);
+            return;
+        }
+        switch (targetState) {
+            case New:
+            case ReInitialized:
+                // do nothing
+                break;
+            case Scheduled:
+                if (currentState == TieringState.New) {
+                    scheduleTableTiering(tableId);
+                } else if (currentState == TieringState.ReInitialized) {
+                    // schedule the table, but we wait for a period of TIMEOUT
+                    // to avoid the existing tiering services are still tiering it since
+                    // it's a table reinitialized by coordinator server
+                    scheduleTableTiering(tableId, TIERING_SERVICE_TIMEOUT_MS);
```

Comment:
We don't need to wait another interval. Because zombie flink job comes back after the timeout, we have to handle the case of two flink job (one is the zombie flink job) both handling the same table. 

We can handle this using epoch mechanism. There are 2 epochs, one is the coordinator epoch which is already there. The other is the table epoch which is increased for every adding into tiering queue. 

The heartbeat report and finish report should check both epoch is equal, otherwise, should response errors.

Reply:
I was intending to not handle two flink jobs are writing paimon files, but do conflict detect while two flink jobs commiting snapshot to Fluss.

It sounds good to me to introduce a table tiering epoch to make tiering service fail fast, so that the tiering writing for zombie flink job can be refused to save resouces(flink resources, file resouces).